src/lib/resolvers/emoji.ts

```diff
@@ -0,0 +1,24 @@
+import { EmojiRegex, TwemojiRegex } from '@sapphire/discord-utilities';
+import { container } from '@sapphire/pieces';
+import { Identifiers } from '../errors/Identifiers';
+import { err, ok } from '../parsers/Result';
+
+export function resolveEmoji(parameter: string) {
+	const twemoji = TwemojiRegex.exec(parameter)?.[0] ?? null;
+
+	if (twemoji) {
+		return ok(twemoji);
+	}
+
+	const emojiId = EmojiRegex.exec(parameter)?.groups?.id;
+
+	if (emojiId) {
+		const resolved = container.client.emojis.resolve(emojiId);
+
+		if (resolved) {
+			return ok(resolved);
+		}
```

Comment:
I *really* dislike that we depend on a slow getter and cache to return an emoji... Ideally we should create an object, ala discord.js's Util.parseEmoji or w/e for it.